src/BlueSCSI.cpp

```diff
@@ -1354,6 +1358,93 @@ byte onTestUnitReady()
   }
   return SCSI_STATUS_GOOD;
 }
+/*
+ * ReZero Unit - Move to Logical Block Zero in file.
+ */
+byte onReZeroUnit() {
+  LOGN("-ReZeroUnit");
+  // Make sure we have an image with atleast a first byte.
+  // Actually seeking to the position wont do anything, so dont.
+  return checkBlockCommand(0, 0);
+}
+
+/*
+ * WriteBuffer - Used for testing buffer, no change to medium
+ */
+byte onWriteBuffer(byte mode, uint32_t allocLength)
+{
+  LOGN("-WriteBuffer");
+  LOGHEXN(mode);
+  LOGHEXN(allocLength);
+
+  if ((mode == MODE_COMBINED_HEADER_DATA || mode == MODE_DATA) && allocLength <= sizeof(m_scsi_buf))
```

Comment:
I think in MODE_COMBINED_HEADER_DATA the allocLength will include the four bytes of header?  Would need to subtract 4 from sizeof(m_scsi_buf) in the combined case.

Also consider refactoring such that the size of m_scsi_buf is set (with perhaps a #define) to avoid the sizeof calculation.